### PR TITLE
PT-3398 Add go to checkout button

### PR DESCRIPTION
**Pull request by ksavosteev**
Description: 
Create a cart with sample data and navigate to checkout.

--

QA-test: PT-3399

Demo-test: 

Download artifact URL: ghcr.io/virtocommerce/demo-xapi-app:1.0.0-pr-15-pt-3210-4b00da4f

### CODE REVIEW

File: src/app/components/navigation-button/navigation-button.component.ts [hunk 1]
```diff
@@ -8,4 +18,51 @@ import { Component } from '@angular/core';
   ],
 })
 export class NavigationButtonComponent {
+  constructor(
+    private readonly apollo: Apollo,
+    private readonly router: Router
+  ) {}
+
+  openCheckout(): void {
+    this.apollo.query<me>({ query: getMeQuery }).subscribe(me => {
```
Comment (asvishnyakov) [inline src/app/components/navigation-button/navigation-button.component.ts]: `unsubsribe` is missed. You may also use `takeeUntil` pattern: https://github.com/VirtoCommerce/vc-demo-xapi-app/blob/1fc7155549d0574bdd50af51472525809c178bbb/src/app/modules/registration/components/personal-information/personal-information.component.ts#L48

File: src/app/modules/checkout/checkout.component.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  routeWatcher!: Subscription;
+
+  constructor(private readonly route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    this.routeWatcher = this.route
+      .queryParams
+      .subscribe(params => {
+        this.cartId = params.cartId as string;
```
Comment (asvishnyakov) [inline src/app/modules/checkout/checkout.component.ts]: `ActivatedRoute` has `queryParamMap` https://angular.io/api/router/ActivatedRoute#queryParamMap which will return typed (`string`) version https://angular.io/api/router/ParamMap

File: src/app/components/navigation-button/navigation-button.component.ts [hunk 3]
```diff
@@ -7,5 +19,68 @@ import { Component } from '@angular/core';
...
-export class NavigationButtonComponent {
+export class NavigationButtonComponent implements OnDestroy {
+  unsubscriber: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private readonly apollo: Apollo,
+    private readonly router: Router
+  ) {}
+
+  openCheckout(): void {
```
Comment (Vectorfield4) [inline src/app/components/navigation-button/navigation-button.component.ts:31]: Extract GraphQL requests into Obeservalbe functions and make one pipe with `concatMap`s and `tap`.
Pseudocode:
```
apolo.query.pipe(
   concatMap(result1 => clearCart(result1.data)),
   concatMap(result2 => addItemsCartMutation(result2.data)),
   tap(result3 => navigate())
)
```

**Review by Vectorfield4 — DISMISSED** [review]

File: src/app/components/navigation-button/navigation-button.component.ts [hunk 4]
```diff
@@ -1,4 +1,16 @@
...
+
+import { clearCart, clearCartVariables } from 'src/app/graphql/types/clearCart';
+import { addItemsCart, addItemsCartVariables } from 'src/app/graphql/types/addItemsCart';
+import { me } from 'src/app/graphql/types/me';
+
+import clearCartMutation from 'src/app/graphql/mutations/clear-cart.graphql';
+import addItemsCartMutation from 'src/app/graphql/mutations/add-cart-items.graphql';
+import getMeQuery from '../../graphql/queries/get-me.graphql';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
```
Comment (Vectorfield4) [inline src/app/components/navigation-button/navigation-button.component.ts]: Move after apolo client.

**Review by Vectorfield4 — APPROVED** [review]

**Review by asvishnyakov — APPROVED** [review]
For the history: I think it's better to use application state (ngrx)

**Pull request merged**